File: 08.feetWet/10.androidAwareness/app/src/main/java/info/juanmendez/myawareness/service/AppModule.java
package info.juanmendez.myawareness.service;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import info.juanmendez.myawareness.App;

/**
 * Created by Juan Mendez on 1/26/2017.
 * www.juanmendez.info
 * contact@juanmendez.info
 */

@Module
public class AppModule {

    private final App app;

    public AppModule( App app ){
        this.app = app;
    }

    @Provides
    @Singleton
    public App application(){
        return app;
    }

}

File: 09.database_connection/08.object_box/app/src/main/java/info/juanmendez/objectbox/AppModule.kt
package info.juanmendez.objectbox

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val context: Application) {
    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }
}